diff -ruNb medCoupling-9.4.0/src/MEDCoupling_Swig/CMakeLists.txt medCoupling-9.4.0_diff/src/MEDCoupling_Swig/CMakeLists.txt
--- medCoupling-9.4.0/src/MEDCoupling_Swig/CMakeLists.txt	2019-08-13 10:45:58.000000000 +0200
+++ medCoupling-9.4.0_diff/src/MEDCoupling_Swig/CMakeLists.txt	2020-05-19 20:34:03.345069100 +0200
@@ -87,7 +87,7 @@
         swig_ready)
 ENDIF()

-IF(WIN32)
+IF(WIN32 OR APPLE)
   # Under Windows MEDCoupling conflicts wiht medcoupling from PyWrapping
   SET(MEDCouling_target_name MEDCouplingCompat)
   SET_PROPERTY(SOURCE MEDCoupling.i PROPERTY SWIG_MODULE_NAME MEDCouplingCompat)
@@ -109,6 +109,9 @@
   # ${PYTHON_LIBRARIES} not needed see https://www.python.org/dev/peps/pep-0513/#libpythonx-y-so-1
   SWIG_LINK_LIBRARIES(${MEDCouling_target_name} ${PLATFORM_LIBS} medcouplingcpp)
 ENDIF(WIN32)
+IF(APPLE)
+  SET_TARGET_PROPERTIES(_${MEDCouling_target_name} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+ENDIF()
 # ${PYTHON_LIBRARIES} not needed
 SWIG_CHECK_GENERATION(${MEDCouling_target_name})
 SET_SOURCE_FILES_PROPERTIES(MEDCouplingRemapper.i PROPERTIES CPLUSPLUS ON)
@@ -137,6 +140,9 @@
   # ${PYTHON_LIBRARIES} not needed see https://www.python.org/dev/peps/pep-0513/#libpythonx-y-so-1
   SWIG_LINK_LIBRARIES(MEDCouplingRemapper medcouplingremapper)
 ENDIF(WIN32)
+IF(APPLE)
+  SET_TARGET_PROPERTIES(_MEDCouplingRemapper PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+ENDIF()

 IF(WIN32)
   SET_TARGET_PROPERTIES(_MEDCouplingRemapper PROPERTIES DEBUG_OUTPUT_NAME _MEDCouplingRemapper_d)
@@ -145,7 +151,7 @@
 INSTALL(TARGETS ${SWIG_MODULE_${MEDCouling_target_name}_REAL_NAME} ${SWIG_MODULE_MEDCouplingRemapper_REAL_NAME} DESTINATION ${MEDCOUPLING_INSTALL_PYTHON})

 SET(PYFILES_TO_INSTALL ${CMAKE_CURRENT_BINARY_DIR}/MEDCouplingRemapper.py)
-IF(WIN32)
+IF(WIN32 OR APPLE)
   SET(PYFILES_TO_INSTALL ${PYFILES_TO_INSTALL} ${CMAKE_CURRENT_BINARY_DIR}/MEDCouplingCompat.py)
 ELSE()
   SET(PYFILES_TO_INSTALL ${PYFILES_TO_INSTALL} ${CMAKE_CURRENT_BINARY_DIR}/MEDCoupling.py)
diff -ruNb medCoupling-9.4.0/src/PyWrapping/CMakeLists.txt medCoupling-9.4.0_diff/src/PyWrapping/CMakeLists.txt
--- medCoupling-9.4.0/src/PyWrapping/CMakeLists.txt	2019-08-13 10:45:58.000000000 +0200
+++ medCoupling-9.4.0_diff/src/PyWrapping/CMakeLists.txt	2020-05-19 20:35:52.230447600 +0200
@@ -94,6 +94,9 @@
 ENDIF()

 SWIG_LINK_LIBRARIES(medcoupling ${medcoupling_LIB_dependancies})
+IF(APPLE)
+  SET_TARGET_PROPERTIES(_medcoupling PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+ENDIF()
 SWIG_CHECK_GENERATION(medcoupling)
 IF(WIN32)
   SET_TARGET_PROPERTIES(_medcoupling PROPERTIES DEBUG_OUTPUT_NAME _medcoupling_d)
